benchmark_utils/dataset.py

```diff
@@ -0,0 +1,76 @@
+from benchopt import safe_import_context
+
+# Protect the import with `safe_import_context()`. This allows:
+# - skipping import to speed up autocompletion in CLI.
+# - getting requirements info when all dependencies are not installed.
+with safe_import_context() as import_ctx:
+    from numpy import multiply
+    from braindecode.preprocessing import (
+        preprocess,
+        Preprocessor,
+    )
+    from braindecode.preprocessing import create_windows_from_events
+
+
+def windows_data(dataset, paradigm_name, trial_start_offset_seconds=-0.5,
+                 low_cut_hz=4.0, high_cut_hz=38.0, factor=1e6, n_jobs=-1,
+                 ):
+    """Create windows from the dataset.
+
+    Parameters:
+    -----------
+    dataset: MOABBDataset
+        Dataset to use.
+    paradigm_name: str
+        Name of the paradigm to use.
+    Returns:
+    --------
+    windows_dataset: WindowsDataset
+        Dataset with windows.
+    sfreq: float
+        Sampling frequency of the dataset.
+    """
+    # Define mapping of classes to integers
+    # We use two classes from the dataset
+    # 1. left-hand vs right-hand motor imagery
+    if paradigm_name == "LeftRightImagery":
+        mapping = {"left_hand": 1, "right_hand": 2}
+
+    elif paradigm_name == "MotorImagery":
+        mapping = {"left_hand": 1, "right_hand": 2, "feet": 4, "tongue": 3}
```

Comment:
isn't there this mapping info directly in the paradigm? this feels weird to hard code this in the function as this week mean we need to maintain a list for every paradigm we want to use

Reply:
at least if we have a list with the babes of the events, we could generate this on the fly

Reply:
This is more of a convention of what we will adopt as left and right order; we can move that somewhere better. We will have to adapt which classes we will use for each dataset and paradigm. The number of classes is always variable per dataset, and no information in the dataset will guarantee alignment and/or order.

I need some help with this point to improve this design.

Reply:
In MOABB this is available in [here](https://github.com/NeuroTechX/moabb/blob/develop/moabb/datasets/bnci.py#L797), I guess similarly it can be accessed through the [mapping](https://github.com/braindecode/braindecode/blob/master/braindecode/preprocessing/windowers.py#L119) in braindecode/mne as well... Maybe it could be use these?